File: aggregator/spring-cloud-stream-aggregator/src/main/java/com/uuhnaut69/jaggregator/event/RestaurantWithDishes.java
package com.uuhnaut69.jaggregator.event;

import com.fasterxml.jackson.annotation.JsonAutoDetect;

import java.util.HashSet;
import java.util.Set;

@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class RestaurantWithDishes {

  private Restaurant restaurant;

  private Set<Dish> dishes = new HashSet<>();

  public RestaurantWithDishes addDish(DishWithRestaurant dishWithRestaurant) {
    restaurant = dishWithRestaurant.getRestaurant();
    dishes.add(dishWithRestaurant.getDish());
    return this;
  }

  public RestaurantWithDishes removeDish(DishWithRestaurant dishWithRestaurant) {
    dishes.removeIf(dish -> dish.getId().equals(dishWithRestaurant.getDish().getId()));
    return this;
  }
}

File: aggregator/kotlin-spring-cloud-streams-aggregator/src/main/kotlin/com/uuhnaut69/kcloudstreams/event/RestaurantWithDishes.kt
package com.uuhnaut69.kcloudstreams.event

import com.fasterxml.jackson.databind.PropertyNamingStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming

@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy::class)
class RestaurantWithDishes {

    var restaurant: Restaurant? = null

    var dishes: MutableSet<Dish> = mutableSetOf()

    fun addDish(dishWithRestaurant: DishWithRestaurant): RestaurantWithDishes {
        restaurant = dishWithRestaurant.restaurant
        dishes.add(dishWithRestaurant.dish)
        return this
    }

    fun removeDish(dishWithRestaurant: DishWithRestaurant): RestaurantWithDishes {
        dishes.removeIf { it.id == dishWithRestaurant.dish.id }
        return this
    }
}